bionic/deriver.py

```diff
@@ -262,7 +257,7 @@ def _compute_result_group_for_dnode(self, dnode):
 
         ready_task_states = list(requested_task_states)
 
-        blocked_task_key_tuples = set()
+        blockage_tracker = BlockageTracker()
```

Comment:
How do you feel about extending this tracker to track ready task states as well?

Reply:
That's a great question. I was planning to do this later, by breaking out a separate `FlowExecution` class, whose lifetime is confined to a single `get` call, and then refactoring it and merging `BlockageTracker` into it. The reason I did it in two stages was we definitely need to get rid of `children`, but I'm not yet 100% sure the full refactor will be a net improvement. On the other hand, the full refactor will probably restructure the blockage tracker stuff, so what you see here is kind of a transitional state.